storage/simulation/main.go

```diff
@@ -18,47 +18,106 @@
 package main
 
 import (
+	"flag"
 	"fmt"
+	"io"
 	"os"
+	"path/filepath"
 
+	"github.com/cockroachdb/cockroach/util/log"
 	"github.com/cockroachdb/cockroach/util/stop"
 )
 
+var maxEpoch = flag.Int("maxEpoch", 10000, "Maximum epoch to simulate.")
+var actionOutputFile = flag.String("action", "", "Output file that shows all actions taken in cluster.")
+var epochOutputFile = flag.String("epoch", "", "Output file that shows stats for all epochs.")
+var startingNodes = flag.Int("startingNodes", 3, "Number of initial nodes in the cluster.")
+var scriptInputFile = flag.String("script", "default.script", "Input script file to describe simulated actions.")
+
 func main() {
 	stopper := stop.NewStopper()
 	defer stopper.Stop()
 
+	flag.Parse()
+	// Give the flags some boundaries.
+	if *startingNodes < 0 {
+		*startingNodes = 0
+	}
+	if *maxEpoch < 0 {
+		*maxEpoch = 0
+	}
+
+	// Clean the output file strings so they can be compared easily or set them
+	// to nil if there is no file.
+	if len(*actionOutputFile) > 0 {
+		*actionOutputFile = filepath.Clean(*actionOutputFile)
+	} else {
+		actionOutputFile = nil
+	}
+	if len(*epochOutputFile) > 0 {
+		*epochOutputFile = filepath.Clean(*epochOutputFile)
+	} else {
+		epochOutputFile = nil
+	}
+	*scriptInputFile = filepath.Clean(*scriptInputFile)
+
 	fmt.Printf("A simulation of the cluster's rebalancing.\n\n")
+	fmt.Printf("Maximum Epoch for simulation set to %d.\n", *maxEpoch)
+	fmt.Printf("Cluster is starting with %d nodes.\n", *startingNodes)
+	fmt.Printf("Script file is %s\n", *scriptInputFile)
 
-	// TODO(bram): Setup flags to allow filenames for these outputs. Print both
-	// to action and epoch to os.Stdout as well.
-	epochWriter := os.Stdout
-	actionWriter := os.Stdout
-	c := createCluster(stopper, 5, epochWriter, actionWriter)
+	epochWriter := io.MultiWriter(os.Stdout)
+	actionWriter := io.MultiWriter(os.Stdout)
 
-	// Split a random range 100 times.
-	for i := 0; i < 10; i++ {
-		c.splitRangeRandom()
+	// Do we have an action output file?
+	if actionOutputFile != nil && len(*actionOutputFile) > 0 {
+		actionOutputF, err := os.OpenFile(*actionOutputFile, os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0666)
```

Comment:
It's actually 0664 there.  I'll create my files with the same permission (and I've reordered the flags to match too).